https://bugs.gentoo.org/903554
https://github.com/WebKit/WebKit/pull/11910
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=109247
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=109307

From 0aa2fbaeb7cd103cda9907f8fa2248cd2209b22f Mon Sep 17 00:00:00 2001
From: Martin Liska <mliska@suse.cz>
Date: Fri, 24 Mar 2023 10:11:51 +0100
Subject: [PATCH] gcc13: fix rejected code by using of an explicit ctor

The following code is invalid based on the analysis here:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=109247#c5
--- a/Source/WebCore/platform/graphics/SourceBrush.cpp
+++ b/Source/WebCore/platform/graphics/SourceBrush.cpp
@@ -65,12 +65,12 @@
 
 void SourceBrush::setGradient(Ref<Gradient>&& gradient, const AffineTransform& spaceTransform)
 {
-    m_brush = { Brush::LogicalGradient { WTFMove(gradient), spaceTransform } };
+    m_brush = Brush { Brush::LogicalGradient { WTFMove(gradient), spaceTransform } };
 }
 
 void SourceBrush::setPattern(Ref<Pattern>&& pattern)
 {
-    m_brush = { WTFMove(pattern) };
+    m_brush = Brush { WTFMove(pattern) };
 }
 
 WTF::TextStream& operator<<(TextStream& ts, const SourceBrush& brush)

